src/studio/src/designer/frontend/app-development/layout/AppBar.tsx

```diff
@@ -0,0 +1,230 @@
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom';
+import { altinnImgLogoHeaderUrl } from 'app-shared/utils/urlHelper';
+import type { IMenuItem } from 'app-shared/navigation/drawer/drawerMenuSettings';
+import TabletDrawerMenu from 'app-shared/navigation/drawer/TabletDrawerMenu';
+import { menu } from './appBarConfig';
+import ProfileMenu from 'app-shared/navigation/main-header/profileMenu';
+
+export interface IAppBarProps {
+  activeSubHeaderSelection?: string;
+  activeLeftMenuSelection?: string;
+  logoutButton?: boolean;
+  org?: string;
+  app?: string;
+  user?: string;
+  showSubMenu?: boolean;
+  mainMenuItems?: IMenuItem[];
+  subMenuItems?: { [key: string]: IMenuItem[] };
+}
+export interface IAppBarComponentState {
+  anchorEl?: any;
+  tabletDrawerOpen: boolean;
+}
+
+const styles = (theme: Theme) =>
+  createStyles({
+    root: {
+      flexGrow: 1,
+      zIndex: 1,
+      color: theme.altinnPalette.primary.black,
+    },
+    appBar: {
+      borderBottom: '1px solid',
+      borderBottomColor: '#C9C9C9',
+      color: theme.altinnPalette.primary.black,
+      position: 'fixed',
+    },
+    appBarDashboard: {
+      backgroundColor: theme.altinnPalette.primary.white,
+    },
+    appBarEditor: {
+      backgroundColor: theme.altinnPalette.primary.greyLight,
+    },
+    breadCrumb: {
+      paddingLeft: 30,
+      fontSize: 20,
+    },
+    breadCrumbSubApp: {
+      color: '#0062BA',
+    },
+    paper: {
+      textAlign: 'center',
+    },
+    subHeader: {
+      paddingLeft: 12,
+      paddingRight: 12,
+      paddingBottom: 0,
+      fontSize: 20,
+    },
+    subHeaderLink: {
+      borderBottom: '1px solid',
+      borderBottomColor: 'transparent', // To mitigate the 1 pixel adjustment
+      '&:hover': {
+        borderBottom: '1px solid',
+        borderBottomColor: theme.altinnPalette.primary.blueDark,
+        color: theme.altinnPalette.primary.blueDark,
+      },
+    },
+    subHeaderLinkActive: {
+      borderBottom: '1px solid',
+      borderBottomColor: theme.altinnPalette.primary.blueDark,
+      color: theme.altinnPalette.primary.blueDark,
+      fontWeight: 500,
+      '&:hover': {
+        borderBottom: '1px solid',
+        borderBottomColor: theme.altinnPalette.primary.blueDark,
+        color: theme.altinnPalette.primary.blueDark,
+      },
+    },
+    topRightService: {
+      paddingRight: 22,
+    },
+    aImgStyling: {
+      borderBottom: 'none',
+      '&:hover': {
+        borderBottom: 'none',
+      },
+      '&:active': {
+        borderBottom: 'none',
+      },
+    },
+  });
+
+const useStyles = makeStyles(styles);
+
+export const AppBar = ({
+  activeLeftMenuSelection,
+  activeSubHeaderSelection,
+  logoutButton,
+  org,
+  app,
+  user,
+  mainMenuItems,
+  subMenuItems,
+  showSubMenu,
+}: IAppBarProps) => {
+  const classes = useStyles();
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const hiddenMdUp = useMediaQuery((theme: Theme) =>
```

Comment:
could this be inverted to `visibleMdUp` and instead of `false` it can be `null`.
So that you can replace all ternary  `hiddenMdUp ? null : [...]` with `visibleMdUp && [...]`

Reply:
I would prefer to stick with the ternary operator instead of `&&` as a general rule of thumb. This is because code is often copied around, and using the `&&` operator could be [problematic in certain cases](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx).